app/src/components/guestbook/GuestBookMessageEdit.tsx

```diff
@@ -34,45 +41,93 @@ export default function GuestBookMessageEdit({
+        message: Yup.string().min(2, 'Too short!').max(1000, 'Too long!').required('Required!'),
+    });
 
     return (
-        <Form>
-            <Form.Group className="mb-3" controlId="EditMessage.Author">
-                <Form.Label>Name</Form.Label>
-                <Form.Control
-                    type="text"
-                    placeholder={placeholderName}
-                    defaultValue={message?.author}
-                    onChange={(event) => updateMessage({ author: event.target.value })}
-                />
-            </Form.Group>
-            <Form.Group className="mb-3" controlId="EditMessage.Message">
-                <Form.Label>Message</Form.Label>
-                <Form.Control
-                    as="textarea"
-                    rows={3}
-                    defaultValue={message?.message}
-                    onChange={(event) => updateMessage({ message: event.target.value })}
-                />
-            </Form.Group>
-            <Stack direction="horizontal" gap={3} className="justify-content-end">
-                <div className="my-auto"></div>
-                <ButtonLink variant="secondary" href={cancelHref}>
-                    Cancel
-                </ButtonLink>
-                <Button variant="primary" onClick={onSubmit} disabled={isSubmitLoading}>
-                    {isSubmitLoading ? <Spinner animation="border" size="sm" /> : submitLabel}
-                </Button>
-            </Stack>
-        </Form>
+        <Formik
+            initialValues={initializeValues()}
+            validationSchema={validationSchema}
+            onSubmit={(message, { setSubmitting }) => {
+                try {
+                    if (onEdit && 'messageId' in message){
+                        console.log('edit new message')
+                        onEdit(message as IGuestBookMessage)
+                    } else if (onCreate) {
+                        console.log('create a message')
+                        onCreate(message as NewGuestBookMessage)
+                    }
+                } catch (error) {
+                    console.log('Failed to create message', error);
+                } finally {
+                    setSubmitting(false);
+                }
+            }}
+        >
+            {(props) => (
+                <Form onSubmit={props.handleSubmit}>
+                    <Form.Group className="mb-3" controlId="EditMessage.Author">
+                        <Form.Label>Name</Form.Label>
+                        <Form.Control
+                            name="author"
+                            type="text"
+                            placeholder={placeholderName}
+                            value={props.values.author}
+                            onBlur={props.handleBlur}
+                            onChange={props.handleChange}
+                        />
+                    </Form.Group>
+                    {props.touched.author && props.errors.author && (
```

Comment:
Nice job, it's good UX to check both `touched` and `errors` before showing an error :)